src/main/java/internetshop/controllers/shoppingcart/ShoppingCartController.java

```diff
@@ -20,10 +20,9 @@
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         if (shoppingCartService.getByUserId(USER_ID).isEmpty()) {
-            shoppingCartService.create(new ShoppingCart(new User(USER_ID)));
+            shoppingCartService.create(new ShoppingCart(new User(USER_ID, "NO_NAME")));
         }
-        req.setAttribute("productsInShoppingCart", shoppingCartService
-                .getAllProducts(shoppingCartService.getByUserId(USER_ID).get()));
+        req.setAttribute("shoppingCart", shoppingCartService.getByUserId(USER_ID).get());
```

Comment:
you are showing information only about shoppingCart products, so let's pass it as attribute shoppingCartService.getByUserId(USER_ID).getProducts()

Reply:
I make it because I use shopping cart id. If I set attribute how products, I need set extra attribute shopping id.